### PR TITLE
Update DPS device symmetric key sample, and fix warnings in other sln samples

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: provisioning/Samples/device/SymmetricKeySample/Parameters.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            Required = true,
+            HelpText = "The Id Scope of the DPS instance")]
+        public string IdScope { get; set; }
+
+        [Option(
+            'r',
+            "RegistrationId",
+            Required = true,
+            HelpText = "The Id of the device to register.")]
+        public string RegistrationId { get; set; }
```
Comment (timtay-microsoft) [inline provisioning/Samples/device/SymmetricKeySample/Parameters.cs:23]: This is only a parameter in individual enrollment scenarios, FYI

Comment (drwill-ms) [reply provisioning/Samples/device/SymmetricKeySample/Parameters.cs:23]: Really it ends up being the device Id. Is it more complicated than I realize?

**Comment by timtay-microsoft** [comment]
In the Java repo, I deliberately split the individual and group enrollment samples for symmetric keys. There is just enough different between the scenarios that I felt it made more sense to split. The two sticking points for me were

- The symmetric key is derived for group enrollments and not individual enrollments

- Group enrollments don't have a "registrationId". They just have a desired device Id. Asking for a registration Id doesn't make sense for group enrollments.

For reference, here are the two java samples:
[individual](https://github.com/Azure/azure-iot-sdk-java/blob/master/provisioning/provisioning-samples/provisioning-symmetrickey-individual-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ProvisioningSymmetricKeyIndividualEnrollmentSample.java)
[group](https://github.com/Azure/azure-iot-sdk-java/blob/master/provisioning/provisioning-samples/provisioning-symmetrickey-group-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ProvisioningSymmetricKeyEnrollmentGroupSample.java)

File: provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+            Console.WriteLine($"Initialized for registration Id {security.GetRegistrationID()}");
+
+            Console.WriteLine("Registering with the device provisioning service... ");
+            DeviceRegistrationResult result = await provClient.RegisterAsync().ConfigureAwait(false);
+
+            Console.WriteLine($"Registration status: {result.Status}.");
+            if (result.Status != ProvisioningRegistrationStatusType.Assigned)
+            {
+                return;
```
Comment (barustum) [inline provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs:56]: We return here without telling a user why we stopped the sample. Do we need to print a console message to the user on what the actual status is before returning?

Comment (abhipsaMisra) [inline provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs:56]: nit: we could print a message here before returning

File: provisioning/Samples/device/SymmetricKeySample/SymmetricKeySample.csproj [hunk 3]
```diff
@@ -2,15 +2,16 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <LangVersion>8.0</LangVersion>
+    <RootNamespace>Microsoft.Azure.Devices.Provisioning.Client.Samples</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="..\Common\ProvisioningDeviceClientSample.cs" Link="ProvisioningDeviceClientSample.cs" />
```
Comment (abhipsaMisra) [inline provisioning/Samples/device/SymmetricKeySample/SymmetricKeySample.csproj:8]: The (provision with dps + test with IoT hub) lived in this common code, with these individual samples showing how to create the different types of security providers. Do we want to deviate from that approach?

**Review by barustum — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: provisioning/Samples/device/SymmetricKeySample/Parameters.cs (hunk 1 above)
Comment (drwill-ms) [reply provisioning/Samples/device/SymmetricKeySample/Parameters.cs:23]: Okay, I see that for individual enrollments (in the Azure portal), a registration Id can be different than the device Id.

Working on it...

File: provisioning/Samples/device/SymmetricKeySample/SymmetricKeySample.csproj (hunk 3 above)
Comment (drwill-ms) [reply provisioning/Samples/device/SymmetricKeySample/SymmetricKeySample.csproj:8]: Yes, I do. Having all the extra code about non-symmetric key for a symmetric key sample is distracting.

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**